src/main/java/duke/logic/parsers/Parser.java

```diff
@@ -0,0 +1,139 @@
+import duke.logic.commands.StaticMapCommand;
+import duke.logic.commands.ViewScheduleCommand;
+import duke.commons.MessagesPrompt;
+import duke.commons.exceptions.DukeException;
+import duke.commons.exceptions.DukeUnknownCommandException;
+
+
+/**
+ * Parser for commands entered by the user. It reads from standard input and
+ * returns Command objects.
+ */
+public class Parser {
+    /**
+     * Parses the userInput and return a Command object.
+     *
+     * @param input Input created by the ConversationManager object or user input.
+     * @return The corresponding Command object.
+     * @throws DukeException If userInput is undefined.
+     */
+    public static Command parseComplexCommand(String input) throws DukeException {
+        String commandWord = getCommandWord(input);
+        switch (commandWord) {
+        case "todo":
+            return new AddCommand(ParserUtil.createTodo(input));
+        case "deadline":
+            return new AddCommand(ParserUtil.createDeadline(input));
+        case "done":
+            return new MarkDoneCommand(ParserUtil.getIndex(input));
+        case "delete":
+            return new DeleteCommand(ParserUtil.getIndex(input));
+        case "find":
+            return new FindCommand(getWord(input));
+        case "findtime":
+            return new FreeTimeCommand(ParserUtil.getIndex(input));
+        case "fetch":
+            return new ViewScheduleCommand(ParserTimeUtil.parseStringToDate(getWord(input)));
+        case "within":
+            return new AddCommand(ParserUtil.createWithin(input));
+        case "reschedule":
+            return new RescheduleCommand(ParserUtil.getSafeIndex(input), ParserUtil.getScheduleDate(input));
+        case "search":
+            return new LocationSearchCommand(getWord(input));
+        case "busStop":
+            return new GetBusStopCommand(getWord(input));
+        case "busRoute":
+            return new GetBusRouteCommand(getWord(input));
+        case "event":
+            return new AddCommand(ParserUtil.createEvent(input));
+        case "findPath":
+            return new FindPathCommand(input.strip().split(" ")[1], getHolidayIndexInList(1, input),
+                    getHolidayIndexInList(2, input));
+        case "recommend":
+            return new RecommendationsCommand(ParserUtil.getIndex(input) + 1);
+        case "cancel":
+            return new PromptCommand(MessagesPrompt.CANCEL_PROMPT);
+        case "map":
+            return new StaticMapCommand(getWord(input));
+        default:
+            throw new DukeUnknownCommandException();
+        }
+    }
+
+    /**
+     * Parses the userInput and return a Command object.
+     *
+     * @param userInput Input created by the ConversationManager object or user input.
+     * @return The corresponding Command object.
+     * @throws DukeException If userInput is undefined.
+     */
+    public static Command parseSingleCommand(String userInput) throws DukeException {
```

Comment:
can you not combine both to one parse? makes usage simpler for anyone writing the client code.